lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1281 @@
+            "Status": "exited"
+        },
+        "StaticDir": "..."
+            ...
+    }'
+"""
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.podman.common import run_podman_command
+
+
+def construct_command_from_params(action, params):
+    if action == 'start':
+        return ['start', params['name']]
+    if action == 'stop':
+        return ['stop', params['name']]
+    if action == 'delete':
+        return ['rm', '-f', params['name']]
+    if action in ['create', 'run']:
+        cmd = ['run', '--name', params['name']]
+
+        if params['detach']:
+            cmd += ['--detach']
+
+        for host_ip in params['etc_hosts'].items():
+            cmd += ['--add-host', ':'.join(host_ip)]
+
+        for annotate in params['annotation'].items():
+            cmd += ['--annotation=' + '='.join(annotate)]
+        # Another thousand parameters
+
+        if params['blkio_weight'] is not None:
+            cmd += ['--blkio-weight', params['blkio_weight']]
+
+        if params['blkio_weight_device']:
+            cmd += ['--blkio-weight-device', params['blkio_weight_device']]
+
+        for cap_add in params['cap_add']:
+            cmd += ['--cap-add', cap_add]
+
+        for cap_drop in params['cap_drop']:
+            cmd += ['--cap-drop', cap_drop]
+
+        if params['cgroup_parent']:
+            cmd += ['--cgroup-parent', params['cgroup_parent']]
+
+        if params['cidfile']:
+            cmd += ['--cidfile', params['cidfile']]
+
+        if params['conmon_pidfile']:
+            cmd += ['--conmon-pidfile', params['conmon_pidfile']]
+
+        if params['cpu_period']:
+            cmd += ['--cpu-period', params['cpu_period']]
+
+        if params['cpu_rt_runtime'] is not None:
+            cmd += ['--cpu-rt-runtime', params['cpu_rt_runtime']]
+
+        if params['cpu_shares'] is not None:
+            cmd += ['--cpu-shares', params['cpu_shares']]
+
+        if params['cpus']:
+            cmd += ['--cpus', params['cpus']]
+
+        if params['cpuset_cpusr']:
+            cmd += ['--cpuset-cpusr', params['cpuset_cpusr']]
+
+        if params['cpuset_mems']:
+            cmd += ['--cpuset-mems', params['cpuset_mems']]
+
+        if params['detach_keys']:
+            cmd += ['--detach-keys', params['detach_keys']]
+
+        if params['device']:
+            cmd += ['--device', params['device']]
+
+        if params['device_read_bps']:
+            cmd += ['--device-read-bps', params['device_read_bps']]
+
+        if params['device_read_iops']:
+            cmd += ['--device-read-iops', params['device_read_iops']]
+
+        if params['device_write_bps']:
+            cmd += ['--device-write-bps', params['device_write_bps']]
+
+        if params['device_write_iops']:
+            cmd += ['--device-write-iops', params['device_write_iops']]
+
+        if params['dns']:
+            cmd += ['--dns=' + ','.join(params['dns'])]
```

Comment:
It's recommended to use string interpolation over concatenation.
```suggestion
            cmd += ['--dns=%s' % ','.join(params['dns'])]
```

Reply:
fixed